dev/release/00-prepare.sh

```diff
@@ -56,6 +56,14 @@ update_versions() {
   git add configure.ac meson.build
   cd -
 
+  cd "${SOURCE_DIR}/../../csharp"
+  sed -i.bak -E -e \
+    "s/^    <VersionPrefix>.+<\/VersionPrefix>/    <VersionPrefix>${version}<\/VersionPrefix>/" \
+    Directory.Build.props
+  rm -f Directory.Build.props.bak
+  git add Directory.Build.props
+  cd -
+
```

Comment:
Thanks!

How about using empty `VersionSuffix` on release tag? If we use empty `VersionSuffix` on release tag, we don't need to pass `-p:VersionSuffix=''` to build release package.

```diff
diff --git a/csharp/Directory.Build.props b/csharp/Directory.Build.props
index fde2ea81..bc0eaa17 100644
--- a/csharp/Directory.Build.props
+++ b/csharp/Directory.Build.props
@@ -13,7 +13,7 @@
     <Copyright>2018 Apache Software Foundation</Copyright>
     <Company>Apache</Company>
     <VersionPrefix>0.13.0</VersionPrefix>
-    <VersionSuffix>preview</VersionSuffix>
+    <VersionSuffix>SNAPSHOT</VersionSuffix>
   </PropertyGroup>
 
   <PropertyGroup>
diff --git a/csharp/README.md b/csharp/README.md
index 1d8dbf83..1eeec818 100644
--- a/csharp/README.md
+++ b/csharp/README.md
@@ -147,7 +147,7 @@ To build the NuGet package run the following command to build a debug flavor, pr
 
 When building the officially released version run: (see Note below about current `git` repository)
 
-    dotnet pack -c Release -p:VersionSuffix=''
+    dotnet pack -c Release
 
 Which will build the final/stable package.
 
diff --git a/dev/release/00-prepare.sh b/dev/release/00-prepare.sh
index f8a5eddf..27101976 100755
--- a/dev/release/00-prepare.sh
+++ b/dev/release/00-prepare.sh
@@ -29,10 +29,14 @@ update_versions() {
   case ${type} in
     release)
       local version=${base_version}
+      local csharp_version_prefix=${base_version}
+      local csharp_version_suffix=
       local r_version=${base_version}
       ;;
     snapshot)
       local version=${next_version}-SNAPSHOT
+      local csharp_version_prefix=${next_version}
+      local csharp_version_suffix=SNAPSHOT
       local r_version=${base_version}.9000
       ;;
   esac
@@ -57,8 +61,9 @@ update_versions() {
   cd -
 
   cd "${SOURCE_DIR}/../../csharp"
-  sed -i.bak -E -e \
-    "s/^    <VersionPrefix>.+<\/VersionPrefix>/    <VersionPrefix>${version}<\/VersionPrefix>/" \
+  sed -i.bak -E \
+    -e "s/^    <VersionPrefix>.+<\/VersionPrefix>/    <VersionPrefix>${csharp_version_prefix}<\/VersionPrefix>/" \
+    -e "s/^    <VersionSuffix>.*<\/VersionSuffix>/    <VersionSuffix>${csharp_version_suffix}<\/VersionSuffix>/" \
     Directory.Build.props
   rm -f Directory.Build.props.bak
   git add Directory.Build.props
```

We don't use any suffix on release tag. For example, https://github.com/apache/arrow/blob/apache-arrow-0.12.1/cpp/CMakeLists.txt#L21 on apache-arrow-0.12.1 tag.

We use "SNAPSHOT" suffix on non release tag. For example, https://github.com/apache/arrow/blob/74436f07c482c0d46a8a52c2b50dc9ad1191922a/cpp/CMakeLists.txt#L21 on the current master.

How about using "SNAPSHOT" instead of "preview" in C# too?

Reply:
> How about using "SNAPSHOT" instead of "preview" in C# too?

Good call. I am used to using `preview` on my team, but since we've already established `SNAPSHOT` in this project, I will make the name change.

> How about using empty VersionSuffix on release tag?

When I started this work, I didn't know how the versioning was handled in this project. But what you describe above makes sense to me. My concerns when I started were:

1. When a dev builds any given commit on their machine, they don't get a "stable" version NuGet package.
2. You shouldn't need to change source code in order to build a "stable" package.

However, I now learned that we already have a process for switching out all the version numbers right before "tagging" for a release. This approach makes sense to me. That way if someone pulls the source code for a release and just "builds", they will get a "stable" package, because they built from the tagged commit for that release. I'll make this change, thanks for the feedback.

Reply:
OK.
It's resonable.